File: Problem statements/week3_greedy_algorithms/1_money_change/change.cpp
#include <iostream>


template<typename T>
T get_change_temp(T&& val)
{
	

};


int get_change(int m) {
	int numberCoins{0};
	while(m >= 10){
		m-=10;
		++numberCoins;
	}
	while(m >= 5){
		m-=5;
		++numberCoins;
	}
	while(m >= 1){
		m-=1;
		++numberCoins;
	}
	return numberCoins;
  //write your code here
//  return n;
}

int main() {
  int m;
  std::cin >> m;
  std::cout << get_change(m) << '\n';
}

File: Problem statements/week3_greedy_algorithms/1_money_change/change.py
# Uses python3
import sys

def get_change(m):
    amount = 0
    while(m > 9):
        amount = amount + 1
        m = m -10
    while(m > 4):
        amount = amount + 1
        m = m -5
    while(m != 0):
        amount = amount + 1
        m = m - 1

    #write your code here
    return amount

if __name__ == '__main__':
    m = int(sys.stdin.read())
    print(get_change(m))
